Content.Server/Tabletop/TabletopSystem.cs

```diff
@@ -0,0 +1,188 @@
+﻿using System.Collections.Generic;
+using Content.Server.Tabletop.Components;
+using Content.Shared.GameTicking;
+using Content.Shared.Tabletop;
+using Content.Shared.Tabletop.Events;
+using JetBrains.Annotations;
+using Robust.Server.GameObjects;
+using Robust.Server.Player;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Map;
+using Robust.Shared.Maths;
+using DrawDepth = Content.Shared.DrawDepth.DrawDepth;
+
+namespace Content.Server.Tabletop
+{
+    [UsedImplicitly]
+    public partial class TabletopSystem : SharedTabletopSystem
+    {
+        [Dependency] private readonly IMapManager _mapManager = default!;
+        [Dependency] private readonly IEntityManager _entityManager = default!;
+
+        private readonly Dictionary<EntityUid, MapId> _gameSessions = new();
+
+        public override void Initialize()
+        {
+            SubscribeNetworkEvent<TabletopMoveEvent>(OnTabletopMove);
+            SubscribeNetworkEvent<TabletopDraggingPlayerChangedEvent>(OnDraggingPlayerChangedEvent);
+            SubscribeLocalEvent<RoundRestartCleanupEvent>(OnRoundRestartCleanup);
+        }
+
+        /**
+         * <summary>
+         * For a specific user, create a table if it does not exist yet and let the user open a UI window to play it.
+         * </summary>
+         * <param name="user">The user entity for which to open the window.</param>
+         * <param name="table">The entity with which the tabletop game session will be associated.</param>
+         */
+        public void OpenTable(IEntity user, IEntity table)
+        {
+            // Make sure we have a table, and get its map ID
+            EnsureTable(table.Uid);
+            var mapId = GetMapId(table.Uid);
+
+            // Create a camera for the user to use
+            // TODO: set correct coordinates, depending on the piece the game was started from
+            IEntity camera = CreateCamera(user, new MapCoordinates(0, 0, mapId));
+
+            // Tell the client that it has to open a viewport for the tabletop game
+            var playerSession = user.PlayerSession();
+            if (playerSession != null)
+            {
+                // Send a message to the client to open a chess UI window
+                // TODO: use actual title/size from prototype, for now we assume its chess
+                _entityManager.EntityNetManager?.SendSystemNetworkMessage(
+                    new TabletopPlayEvent(table.Uid, camera.Uid, "Chess", (274 + 64, 274)), playerSession.ConnectedClient
```

Comment:
sus that that value is hardcoded

Reply:
Also don't need to go to SendSystemNetworkMessage in this roundabout way when RaiseNetworkEvent exists.

Reply:
Value is hardcoded for now because its always chess, will be fixed when other games are implemented.
SendSystemNetworkMessage is fixed.